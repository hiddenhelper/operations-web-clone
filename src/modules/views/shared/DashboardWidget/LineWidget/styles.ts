import { createStyles, makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles(theme =>
  createStyles({
    widgetTitleMarginBottom: {
      marginBottom: 18,
    },
    revenueWidgetContent: {
      display: 'flex',
      justifyContent: 'space-between',
      flexDirection: 'column',
      alignItems: 'center',
      maxHeight: '350px',
      paddingRight: 0,
      height: 350,
    },
  })
);
