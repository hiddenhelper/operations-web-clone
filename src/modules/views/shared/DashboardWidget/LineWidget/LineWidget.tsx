import React, { memo } from 'react';

import Grid from '@material-ui/core/Grid';

import DashboardWidget from '../../DashboardWidget';
import DashboardWidgetHeader from '../../DashboardWidget/DashboardWidgetHeader';
import LineChart from '../../Chart/LineChart';

import { useStyles } from './styles';

export interface ILineWidgetProps {
  title: string;
  total: number | string;
  data: { x: string; y: number }[];
  isLoading: boolean;
  padding?: { left: number; right: number; bottom: number; top: number };
  displayXAxisTicks?: (tick: any) => void;
}

const LineWidget = ({ title, total, data, padding = { left: 35, right: 10, bottom: 40, top: 30 }, isLoading, displayXAxisTicks }: ILineWidgetProps) => {
  const classes = useStyles();
  return (
    <DashboardWidget
      renderHeader={() => (
        <div className={`${classes.widgetTitleMarginBottom}`}>
          <DashboardWidgetHeader title={title} value={total} isLoading={isLoading} />
        </div>
      )}
      renderContent={() => (
        <Grid
          container={true}
          style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            minHeight: '335px',
          }}
        >
          <Grid item={true} xl={12} lg={12} md={12} xs={12} style={{ maxHeight: '370px', height: '370px' }} className={classes.revenueWidgetContent}>
            <LineChart
              width={600}
              height={530}
              isLoading={isLoading}
              fontSize={16}
              padding={padding}
              displayXAxisTicks={displayXAxisTicks}
              series={[
                {
                  name: 'serie 1',
                  data: data,
                  style: {
                    data: {
                      stroke: '#1F86E8',
                      strokeWidth: 3,
                    },
                  },
                },
              ]}
            />
          </Grid>
        </Grid>
      )}
    />
  );
};

export default memo(LineWidget);
