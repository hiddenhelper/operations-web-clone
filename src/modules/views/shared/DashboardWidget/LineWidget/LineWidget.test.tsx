import React from 'react';
import { render } from '@testing-library/react';

import LineWidget, { ILineWidgetProps } from './LineWidget';

describe('LineWidget', () => {
  let props: ILineWidgetProps;

  beforeEach(() => {
    props = {
      title: 'title',
      total: 1,
      data: [{ x: 'x axis', y: 12 }],
      isLoading: false,
    };
  });

  it('should render', () => {
    const { container } = render(<LineWidget {...props} />);
    expect(container).toMatchSnapshot();
  });
});
