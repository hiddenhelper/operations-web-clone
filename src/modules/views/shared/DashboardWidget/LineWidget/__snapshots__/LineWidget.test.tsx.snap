// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineWidget should render 1`] = `
<div>
  <div
    class="MuiPaper-root  makeStyles-widget-3 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      class="makeStyles-widgetTitleMarginBottom-1"
    >
      <p
        class="MuiTypography-root makeStyles-widgetTitle-5 makeStyles-widgetTitleBottomMargin-9 MuiTypography-body1"
      >
        title
      </p>
      <div
        class="makeStyles-widgetContentWrapper-10"
      >
        <span
          class="makeStyles-widgetContentResponsive-16 makeStyles-widgetContent-15"
        >
          1
        </span>
        <div
          class="makeStyles-chipStatusLeftMargin-25"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container"
      style="display: flex; justify-content: center; align-items: center; min-height: 335px;"
    >
      <div
        class="MuiGrid-root makeStyles-revenueWidgetContent-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        style="max-height: 370px; height: 370px;"
      >
        <div
          class="VictoryContainer"
          style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
        >
          <svg
            height="530"
            role="img"
            style="pointer-events: all; width: 100%; height: 100%;"
            viewBox="0 0 600 530"
            width="600"
          >
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; stroke-width: 0; fill: transparent; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="35"
                x2="35"
                y1="30"
                y2="490"
              />
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: #ECEFF1; stroke-dasharray: 1; stroke-width: 1; fill: none; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="590"
                  y1="490"
                  y2="490"
                />
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: white; size: 0; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="35"
                  y1="490"
                  y2="490"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-axis-0-tickLabels-0"
                  x="27"
                  y="495.68"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-size: 16px; fill: #878787; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; letter-spacing: normal; padding: 8px; stroke: transparent; stroke-width: 0;"
                    text-anchor="end"
                    x="27"
                  />
                </text>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: #ECEFF1; stroke-dasharray: 1; stroke-width: 1; fill: none; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="590"
                  y1="374.99795718980374"
                  y2="374.99795718980374"
                />
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: white; size: 0; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="35"
                  y1="374.99795718980374"
                  y2="374.99795718980374"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-axis-0-tickLabels-1"
                  x="27"
                  y="380.67795718980375"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-size: 16px; fill: #878787; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; letter-spacing: normal; padding: 8px; stroke: transparent; stroke-width: 0;"
                    text-anchor="end"
                    x="27"
                  />
                </text>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: #ECEFF1; stroke-dasharray: 1; stroke-width: 1; fill: none; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="590"
                  y1="260"
                  y2="260"
                />
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: white; size: 0; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="35"
                  y1="260"
                  y2="260"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-axis-0-tickLabels-2"
                  x="27"
                  y="265.68"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-size: 16px; fill: #878787; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; letter-spacing: normal; padding: 8px; stroke: transparent; stroke-width: 0;"
                    text-anchor="end"
                    x="27"
                  >
                    12
                  </tspan>
                </text>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: #ECEFF1; stroke-dasharray: 1; stroke-width: 1; fill: none; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="590"
                  y1="145.00204281019626"
                  y2="145.00204281019626"
                />
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: white; size: 0; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="35"
                  y1="145.00204281019626"
                  y2="145.00204281019626"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-axis-0-tickLabels-3"
                  x="27"
                  y="150.68204281019626"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-size: 16px; fill: #878787; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; letter-spacing: normal; padding: 8px; stroke: transparent; stroke-width: 0;"
                    text-anchor="end"
                    x="27"
                  />
                </text>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: #ECEFF1; stroke-dasharray: 1; stroke-width: 1; fill: none; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="590"
                  y1="30"
                  y2="30"
                />
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: white; size: 0; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  vector-effect="non-scaling-stroke"
                  x1="35"
                  x2="35"
                  y1="30"
                  y2="30"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-axis-0-tickLabels-4"
                  x="27"
                  y="35.68"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-size: 16px; fill: #878787; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; letter-spacing: normal; padding: 8px; stroke: transparent; stroke-width: 0;"
                    text-anchor="end"
                    x="27"
                  />
                </text>
              </g>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #eceff1; stroke-width: 1; fill: transparent; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="35"
                x2="590"
                y1="490"
                y2="490"
              />
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: #ECEFF1; stroke-dasharray: 1; stroke-width: 1; display: none; fill: none; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                  vector-effect="non-scaling-stroke"
                  x1="312.5"
                  x2="312.5"
                  y1="490"
                  y2="30"
                />
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: white; size: 0; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  vector-effect="non-scaling-stroke"
                  x1="312.5"
                  x2="312.5"
                  y1="490"
                  y2="490"
                />
                <text
                  direction="inherit"
                  dx="15"
                  id="chart-axis-1-tickLabels-0"
                  x="312.5"
                  y="511.68"
                >
                  <tspan
                    dx="15"
                    dy="0"
                    style="font-size: 16px; fill: #878787; dominant-baseline: hanging; display: block; font-family: 'Helvetica Neue', 'Helvetica', sans-serif; letter-spacing: normal; padding: 8px; stroke: transparent; stroke-width: 0;"
                    text-anchor="middle"
                    x="312.5"
                  >
                    x axis
                  </tspan>
                </text>
              </g>
            </g>
            <g
              clip-path="url(#victory-clip-2)"
              style="height: 100%; width: 100%; user-select: none;"
            >
              <defs>
                <clippath
                  id="victory-clip-2"
                >
                  <rect
                    height="460"
                    vector-effect="non-scaling-stroke"
                    width="555"
                    x="35"
                    y="30"
                  />
                </clippath>
              </defs>
              <path
                role="presentation"
                shape-rendering="auto"
                style="fill: transparent; stroke: #1F86E8; stroke-width: 3; opacity: 1;"
              />
            </g>
          </svg>
          <div
            style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
          >
            <svg
              height="530"
              style="overflow: visible; width: 100%; height: 100%;"
              viewBox="0 0 600 530"
              width="600"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
